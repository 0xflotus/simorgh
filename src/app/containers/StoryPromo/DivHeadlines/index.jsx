import styled from 'styled-components';
import { bool, string, oneOf, shape } from 'prop-types';
import { GEL_SPACING } from '@bbc/gel-foundations/spacings';
import { C_EBON } from '@bbc/psammead-styles/colours';
import { getSerifMedium } from '@bbc/psammead-styles/font-styles';
import {
  getParagon,
  getPica,
  getDoublePica,
} from '@bbc/gel-foundations/typography';
import { scriptPropType } from '@bbc/gel-foundations/prop-types';

const PROMO_TYPES = oneOf(['top', 'regular', 'leading']);

const headlineTopStoryTypography = script => getParagon(script);

const headlineRegularTypography = script => getPica(script);

const headlineLeadingStoryTypography = script => getDoublePica(script);

const headlineTypography = script => ({
  top: headlineTopStoryTypography(script),
  regular: headlineRegularTypography(script),
  leading: headlineLeadingStoryTypography(script),
});

const DivHeadline = styled.div`
  color: ${C_EBON};
  margin: 0; /* Reset */
  padding-bottom: ${GEL_SPACING};
  ${({ service }) => getSerifMedium(service)}
  ${({ script, promoType }) => script && headlineTypography(script)[promoType]}
  ${({ promoHasImage }) =>
    !promoHasImage &&
    `display: inline;`} /* Needed for aligning Media Indicator with Headline */
`;

DivHeadline.propTypes = {
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
  promoHasImage: bool,
  promoType: PROMO_TYPES,
};

DivHeadline.defaultProps = {
  promoHasImage: true,
  promoType: 'regular',
};

export default DivHeadline;
