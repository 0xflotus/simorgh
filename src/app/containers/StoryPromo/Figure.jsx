import React from 'react';
import { storyItemImage } from '../../models/propTypes/storyItem';
import Figure from '../Figure';

const StoryPromoFigure = ({ path, altText, height, width }) => {
  if (!path || !altText || !width || !height) {
    return null;
  }

  const ratio = ((height / width) * 100).toFixed(4);

  return (
    <Figure
      alt={altText}
      ratio={ratio}
      src={`https://ichef.bbci.co.uk/news/660${path}`}
      height={height}
      width={width}
    />
  );
};

StoryPromoFigure.propTypes = storyItemImage;

export default StoryPromoFigure;
