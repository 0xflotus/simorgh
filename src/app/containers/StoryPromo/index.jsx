import React, { Fragment, useContext } from 'react';
import { shape, bool, string, element } from 'prop-types';
import StoryPromoComponent, {
  Headline,
  Summary,
  Link,
  LiveLabel,
} from '@bbc/psammead-story-promo';
import Timestamp from '@bbc/psammead-timestamp-container';
import pathOr from 'ramda/src/pathOr';
import VisuallyHiddenText from '@bbc/psammead-visually-hidden-text';
import { storyItem } from '../../models/propTypes/storyItem';
import ImageWithPlaceholder from '../ImageWithPlaceholder';

import { ServiceContext } from '../../contexts/ServiceContext';
import createSrcset from '../Image/helpers/srcSet';
import getOriginCode from './imageSrcHelpers/originCode';
import getLocator from './imageSrcHelpers/locator';

import LinkContents from './LinkContents';
import MediaIndicator from './MediaIndicator';
import isTenHoursAgo from '../../lib/utilities/isTenHoursAgo';

const StoryPromoImage = ({ topStory, imageValues, lazyLoad }) => {
  if (!imageValues) {
    return null;
  }

  const { height, width, path } = imageValues;

  const ratio = (height / width) * 100;
  const originCode = getOriginCode(path);
  const locator = getLocator(path);
  const imageResolutions = [70, 95, 144, 183, 240, 320, 480, 624];
  const srcset = createSrcset(originCode, locator, width, imageResolutions);
  const sizes = topStory
    ? '(max-width: 600px) 100vw, (max-width: 1008px) 33vw, 237px'
    : '(max-width: 1008px) 33vw, 237px';
  const DEFAULT_IMAGE_RES = 660;
  const src = `https://ichef.bbci.co.uk/news/${DEFAULT_IMAGE_RES}${path}`;

  return (
    <ImageWithPlaceholder
      alt={imageValues.altText}
      ratio={ratio}
      src={src}
      fallback={false}
      {...imageValues}
      lazyLoad={lazyLoad}
      copyright={imageValues.copyrightHolder}
      srcset={srcset}
      sizes={sizes}
    />
  );
};

StoryPromoImage.propTypes = {
  lazyLoad: bool.isRequired,
  topStory: bool.isRequired,
  imageValues: shape(storyItem.indexImage).isRequired,
};

const LiveComponent = ({ headline, service, dir }) => (
  // eslint-disable-next-line jsx-a11y/aria-role
  <span role="text">
    <LiveLabel service={service} dir={dir}>
      LIVE
    </LiveLabel>
    <VisuallyHiddenText lang="en-GB">Live, </VisuallyHiddenText>
    {headline}
  </span>
);

LiveComponent.propTypes = {
  service: string.isRequired,
  dir: string.isRequired,
  headline: element.isRequired,
};

const StoryPromo = ({ item, lazyLoadImage, topStory }) => {
  const { script, datetimeLocale, service, dir } = useContext(ServiceContext);
  const headline = pathOr(null, ['headlines', 'headline'], item);
  const url = pathOr(null, ['locators', 'assetUri'], item);
  const summary = pathOr(null, ['summary'], item);
  let timestamp = pathOr(null, ['timestamp'], item);
  const isLive = pathOr(null, ['cpsType'], item) === 'LIV';

  if (new Date(timestamp).getFullYear() < 1980) {
    // if the date is before 1980, our timestamp was probably in seconds.
    // this fixes an ares bug - ARES-758 on JIRA.
    // if you come across this in the future, please check if it's no longer needed
    // if so, delete this!
    timestamp *= 1000;
  }

  const linkcontents = <LinkContents item={item} />;

  if (!headline || !url) {
    return null;
  }

  const Info = (
    <Fragment>
      {headline && (
        <Headline script={script} service={service} topStory={topStory}>
          <Link href={url}>
            {isLive ? (
              <LiveComponent
                service={service}
                headline={linkcontents}
                dir={dir}
              />
            ) : (
              linkcontents
            )}
          </Link>
        </Headline>
      )}
      {summary && (
        <Summary script={script} service={service} topStory={topStory}>
          {summary}
        </Summary>
      )}
      {timestamp && (
        <Timestamp
          locale={datetimeLocale}
          timestamp={timestamp}
          dateTimeFormat="YYYY-MM-DD"
          format="D MMMM YYYY"
          script={script}
          padding={false}
          service={service}
          isRelative={isTenHoursAgo(timestamp)}
        />
      )}
    </Fragment>
  );

  const imageValues = pathOr(null, ['indexImage'], item);
  const Image = (
    <StoryPromoImage
      topStory={topStory}
      lazyLoad={lazyLoadImage}
      imageValues={imageValues}
    />
  );

  return (
    <StoryPromoComponent
      image={Image}
      info={Info}
      mediaIndicator={
        <MediaIndicator item={item} topStory={topStory} service={service} />
      }
      topStory={topStory}
    />
  );
};

StoryPromo.propTypes = {
  item: shape(storyItem).isRequired,
  lazyLoadImage: bool,
  topStory: bool,
};

StoryPromo.defaultProps = {
  lazyLoadImage: true,
  topStory: false,
};

export default StoryPromo;
