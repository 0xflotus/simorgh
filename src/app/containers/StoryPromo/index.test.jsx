import React from 'react';
import { render } from 'react-testing-library';
import { shouldMatchSnapshot } from '../../helpers/tests/testHelpers';
import StoryPromo from '.';

let item = {
  headlines: {
    headline: 'A headline',
  },
  summary: 'Summary text',
  timestamp: 1556795033,
  indexImage: {
    path: '/cpsprodpb/0A06/production/image.jpg',
    height: 1152,
    width: 2048,
    altText: 'Image Alt text',
  },
};

describe('StoryPromo Container', () => {
  shouldMatchSnapshot('should render correctly', <StoryPromo item={item} />);

  describe('assertion tests', () => {
    beforeEach(() => {
      item = {
        headlines: {
          headline: 'A headline',
        },
        summary: 'Summary text',
        timestamp: 1556795033,
        indexImage: {
          path: '/cpsprodpb/0A06/production/image.jpg',
          height: 1152,
          width: 2048,
          altText: 'Image Alt text',
        },
      };
    });

    it('should correctly include expected elements', () => {
      const { container } = render(<StoryPromo item={item} />);

      expect(container.getElementsByTagName('h3')[0].innerHTML).toEqual(
        item.headlines.headline,
      );
      expect(container.getElementsByTagName('p')[0].innerHTML).toEqual(
        item.summary,
      );
      expect(container.getElementsByTagName('time')[0].innerHTML).toEqual(
        '2 May 2019',
      );
      expect(container.getElementsByTagName('img')[0].src).toEqual(
        `https://ichef.bbci.co.uk/news/660${item.indexImage.path}`,
      );
      expect(
        container.getElementsByTagName('img')[0].getAttribute('alt'),
      ).toEqual(item.indexImage.altText);
    });

    describe('With no headline provided', () => {
      beforeEach(() => {
        delete item.headlines;
      });

      it('should not include a headline element', () => {
        const { container } = render(<StoryPromo item={item} />);

        expect(container.getElementsByTagName('h3').length).toEqual(0);
      });
    });

    describe('With no summary provided', () => {
      beforeEach(() => {
        delete item.summary;
      });

      it('should not include a paragraph element', () => {
        const { container } = render(<StoryPromo item={item} />);

        expect(container.getElementsByTagName('p').length).toEqual(0);
      });
    });

    describe('With no timestamp provided', () => {
      beforeEach(() => {
        delete item.timestamp;
      });

      it('should not include a time element', () => {
        const { container } = render(<StoryPromo item={item} />);

        expect(container.getElementsByTagName('time').length).toEqual(0);
      });
    });

    describe('With no indexImage provided', () => {
      beforeEach(() => {
        delete item.indexImage;
      });

      it('should not include a time element', () => {
        const { container } = render(<StoryPromo item={item} />);

        expect(container.getElementsByTagName('img').length).toEqual(0);
      });
    });
  });
});
