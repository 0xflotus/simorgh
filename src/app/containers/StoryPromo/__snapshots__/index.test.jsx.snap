// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryPromo Container should render correctly for amp 1`] = `
.c1 {
  display: inline-block;
  vertical-align: top;
  max-width: 33.33%;
  position: relative;
}

.c7 {
  display: inline-block;
  vertical-align: top;
  padding: 0 0.5rem;
  max-width: 66.67%;
}

.c8 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #222222;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
  font-weight: 700;
}

.c10 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #3F3F42;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c9 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c9:hover,
.c9:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:visited {
  color: #6E6E73;
}

.c11 {
  font-size: 0.875rem;
  line-height: 1rem;
  color: #6E6E73;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
}

.c2 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.c5 {
  position: relative;
  height: 0;
  overflow: hidden;
  background-color: #ECEAE7;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: 30%;
  padding-bottom: 56.25%;
  width: 100%;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
}

.c3 {
  display: grid;
  grid-column-gap: 1rem;
}

@media (min-width:25rem) {
  .c0 {
    position: relative;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    max-width: 33.33%;
  }
}

@supports (display:grid) {
  .c1 {
    display: block;
    max-width: initial;
    grid-column: 1 / span 2;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c7 {
    max-width: 66.67%;
  }
}

@supports (display:grid) {
  .c7 {
    display: block;
    max-width: initial;
    padding: initial;
    grid-column: 3 / span 4;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c10 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c10 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (max-width:37.4375rem) {
  .c10 {
    display: none;
    visibility: hidden;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c11 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c11 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

@media (max-width:62.9375rem) {
  .c4 {
    grid-column: 1 / span 6;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c4 {
    grid-column: 1 / span 6;
  }
}

@media (min-width:80rem) {
  .c4 {
    grid-column: 1 / span 12;
  }
}

@media (max-width:62.9375rem) {
  .c6 {
    grid-column: 1 / span 5;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c6 {
    grid-column: 1 / span 2;
  }
}

@media (min-width:80rem) {
  .c6 {
    grid-column: 1 / span 8;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c3 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:80rem) {
  .c3 {
    grid-template-columns: repeat(12,1fr);
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <figure
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <div
            className="c5"
          >
            <amp-img
              alt="Image Alt text"
              attribution="Image provider"
              height={1152}
              layout="responsive"
              src="https://ichef.bbci.co.uk/news/660/cpsprodpb/0A06/production/image.jpg"
              width={2048}
            />
          </div>
        </div>
        <div
          className="c6"
        />
      </div>
    </figure>
  </div>
  <div
    className="c7"
  >
    <h3
      className="c8"
    >
      <a
        className="c9"
        href="https://www.bbc.co.uk"
      >
        A headline
      </a>
    </h3>
    <p
      className="c10"
    >
      Summary text
    </p>
    <time
      className="c11"
      dateTime="2019-05-02"
    >
      2 May 2019
    </time>
  </div>
</div>
`;

exports[`StoryPromo Container should render correctly for canonical 1`] = `
.c1 {
  display: inline-block;
  vertical-align: top;
  max-width: 33.33%;
  position: relative;
}

.c8 {
  display: inline-block;
  vertical-align: top;
  padding: 0 0.5rem;
  max-width: 66.67%;
}

.c9 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #222222;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
  font-weight: 700;
}

.c11 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #3F3F42;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c10 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c10:hover,
.c10:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c10:visited {
  color: #6E6E73;
}

.c12 {
  font-size: 0.875rem;
  line-height: 1rem;
  color: #6E6E73;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
}

.c2 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.c6 {
  display: block;
  width: 100%;
}

.c5 {
  position: relative;
  height: 0;
  overflow: hidden;
  background-color: #ECEAE7;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: 30%;
  padding-bottom: 56.25%;
  width: 100%;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
}

.c3 {
  display: grid;
  grid-column-gap: 1rem;
}

@media (min-width:25rem) {
  .c0 {
    position: relative;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    max-width: 33.33%;
  }
}

@supports (display:grid) {
  .c1 {
    display: block;
    max-width: initial;
    grid-column: 1 / span 2;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c8 {
    max-width: 66.67%;
  }
}

@supports (display:grid) {
  .c8 {
    display: block;
    max-width: initial;
    padding: initial;
    grid-column: 3 / span 4;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c11 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c11 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (max-width:37.4375rem) {
  .c11 {
    display: none;
    visibility: hidden;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c12 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c12 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

@media (max-width:62.9375rem) {
  .c4 {
    grid-column: 1 / span 6;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c4 {
    grid-column: 1 / span 6;
  }
}

@media (min-width:80rem) {
  .c4 {
    grid-column: 1 / span 12;
  }
}

@media (max-width:62.9375rem) {
  .c7 {
    grid-column: 1 / span 5;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c7 {
    grid-column: 1 / span 2;
  }
}

@media (min-width:80rem) {
  .c7 {
    grid-column: 1 / span 8;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c3 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:80rem) {
  .c3 {
    grid-template-columns: repeat(12,1fr);
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <figure
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <div
            className="c5"
          >
            <div
              className="lazyload-placeholder"
              style={
                Object {
                  "height": undefined,
                }
              }
            />
            <noscript>
              <img
                alt="Image Alt text"
                className="c6"
                height={null}
                src="https://ichef.bbci.co.uk/news/660/cpsprodpb/0A06/production/image.jpg"
                width={2048}
              />
            </noscript>
          </div>
        </div>
        <div
          className="c7"
        />
      </div>
    </figure>
  </div>
  <div
    className="c8"
  >
    <h3
      className="c9"
    >
      <a
        className="c10"
        href="https://www.bbc.co.uk"
      >
        A headline
      </a>
    </h3>
    <p
      className="c11"
    >
      Summary text
    </p>
    <time
      className="c12"
      dateTime="2019-05-02"
    >
      2 May 2019
    </time>
  </div>
</div>
`;
